sdcm/cluster.py

```diff
@@ -4190,7 +4190,7 @@ def node_setup(self, node: BaseNode, verbose: bool = False, timeout: int = 3600)
             self._scylla_post_install(node, install_scylla, nic_devname)
 
             # prepare and start saslauthd service
-            if self.params.get('prepare_saslauthd'):
+            if node.is_enterprise and self.params.get('prepare_saslauthd'):
```

Comment:
the `prepare_saslauthd` flag is default to false, so this doesn't help:
the problematic code is here:
```
    @property
    def saslauthd_socket_path(self) -> Optional[str]:
        if self._is_authenticator_valid:
            return '/run/saslauthd/mux'
        return None
```

`_is_authenticator_valid` can be true in lots of cases, regardless if salauthd is enabled or not.

maybe this would fix it:
```
    @property
    def saslauthd_socket_path(self) -> Optional[str]:
        if self._is_authenticator_valid and self.params.get('prepare_saslauthd'):
            return '/run/saslauthd/mux'
        return None
```

Reply:
fixed